internal/controlplane/handlers_oauth.go

```diff
@@ -448,6 +449,39 @@ func (s *Server) VerifyProviderTokenFrom(ctx context.Context,
 	entityCtx := engine.EntityFromContext(ctx)
 	projectID := entityCtx.Project.ID
 
+	// Telemetry logging
+	logger.BusinessRecord(ctx).Provider = in.GetContext().GetProvider()
+	logger.BusinessRecord(ctx).Project = projectID
+
+	// Provider Enroll probes this endpoint to see whether the provider has
+	// been set up.  GitHub Apps don't create a provider row until it the
+	// install for a particular app has been created, so return "not ready"
+	// in the meantime.
+	if in.GetContext().GetProvider() == string(db.ProviderClassGithubApp) {
+		// TODO: factor this into a method someplace better.
+
+		// We don't know which app the user is trying to register, but see if there's
+		// a recently-created one.
+		prov, err := s.store.FindProviders(ctx, db.FindProvidersParams{
+			Projects: []uuid.UUID{projectID},
+			Trait: db.NullProviderType{
+				ProviderType: db.ProviderTypeGithub,
```

Comment:
didn't we have an explicit trait for github apps?

Reply:
No, we just have the class